test/central-config-enabled.js

```diff
@@ -117,3 +119,46 @@ test('remote config enabled: receives non delimited string', function (t) {
 
   runTestsWithServer(t, updates, expect)
 })
+
+// Ensure the logger updates if the central config `log_level` changes.
+test('agent.logger updates for central config `log_level` change', { timeout: 1000 }, function (t) {
+  let agent
+
+  const server = http.createServer((req, res) => {
+    // 3. The agent should fetch central config with log_level=error.
+    const url = new URL(req.url, 'relative:///')
+    t.strictEqual(url.pathname, '/config/v1/agents')
+    res.writeHead(200, {
+      Etag: 1,
+      'Cache-Control': 'max-age=30, must-revalidate'
+    })
+    res.end(JSON.stringify({ log_level: 'error' }))
+
+    const SHORT_TIME_LATER = 10 // Hopefully this isn't a race we lose.
+    setTimeout(function () {
+      // 4. agent.logger should be updated from central config.
+      t.equal(agent.logger.level, 'error',
+        'shortly after fetching central config, agent.logger level should be updated')
+
+      agent.destroy()
+      server.close()
+      t.end()
+    }, SHORT_TIME_LATER)
```

Comment:
Would an event listener (`agent._transport.on('config',`?) be less prone to a race condition than this `setTimeout` function?

Reply:
I'll give it a try, thanks.

Reply:
Much nicer. Thanks! With your suggestion, I'm unable to get that sequencing to fail with a number of dozen quick tries:

```
while true; do if ! node test/central-config-enabled.js; then break; fi; done
```

whereas with the possibly-race setTimeout and setting the timeout to 0ms or 1ms I was able to occasionally  get a failure.